gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -150,6 +150,46 @@ private void handleDeepLinks(Uri data) {
                     Toast.LENGTH_SHORT).show();
             }
     }
+
+    /**
+     * Normalizes the string by removing punctuation, removing whitespaces
+     * and making the string lowercase.
+     * @param input input string
+     * @return a normalized string
+     */
+    public String normalize(String input) {
```

Comment:
Comparing the code with the comment, it seems like there are some cornercases which would be handled unexpected, e.g., what about unicode characters?